### DISCUSSION TITLE
ScriptReference/Collider.ClosestPoint

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#292] [Notes]
# ScriptReference/Collider.ClosestPoint

https://docs.unity3d.com/ScriptReference/Collider.ClosestPoint.html

**Comment by joonjoonjoon** [comment]
The position parameter is returned if it is inside of the collider.

**Comment by toxicaliengf** [comment]
This will return the original Vector3 position when using a Terrain Collider :/

**Comment by cjacobwade** [comment]
This function always returns the passed in Vector3 if the collider is disabled or the gameObject not activeInHierarchy!

**Comment by STARasGAMES** [comment]
This is what you get when you try to use it on MeshCollider:

Physics.ClosestPoint can only be used with a BoxCollider, SphereCollider, CapsuleCollider and a convex MeshCollider.

**Reply by laurentopia** [reply]
Concave mesh colliders need to be CoACD-ed, and when that fails, precompute SDF of the static scene and use distance check. Fun!
